CMakeLists.txt

```diff
@@ -107,48 +107,11 @@ add_subdirectory(NCEPLIBS-nemsiogfs)
-file( APPEND ${SCRIPT_NAME_BASH} "export ESMFMKFILE=${ESMFMKFILE}\n")
-file( APPEND ${SCRIPT_NAME_BASH} "export NETCDF=${NETCDF}\n")
-file( APPEND ${SCRIPT_NAME_BASH} "# NCEPLIBS\n")
-file( APPEND ${SCRIPT_NAME_BASH} "export NCEPLIBS_DIR=${CMAKE_INSTALL_PREFIX}\n")
-file( APPEND ${SCRIPT_NAME_BASH} "export NEMSIO_INC=${CMAKE_INSTALL_PREFIX}/include\n")
-file( APPEND ${SCRIPT_NAME_BASH} "export NEMSIO_LIB=${CMAKE_INSTALL_PREFIX}/lib/libnemsio_v2.2.3.a\n")
-file( APPEND ${SCRIPT_NAME_BASH} "export BACIO_LIB4=${CMAKE_INSTALL_PREFIX}/lib/libbacio_v2.1.0_4.a\n")
-file( APPEND ${SCRIPT_NAME_BASH} "export SP_LIBd=${CMAKE_INSTALL_PREFIX}/lib/libsp_v2.0.2_d.a\n")
-file( APPEND ${SCRIPT_NAME_BASH} "export W3EMC_LIBd=${CMAKE_INSTALL_PREFIX}/lib/libw3emc_v2.2.0_d.a\n")
-file( APPEND ${SCRIPT_NAME_BASH} "export W3NCO_LIBd=${CMAKE_INSTALL_PREFIX}/lib/libw3nco_v2.0.6_d.a\n")
-file( APPEND ${SCRIPT_NAME_BASH} "#\n")
-
-message(STATUS "Creating tcsh shell script ${SCRIPT_NAME_TCSH} to set environment variables for using NCEPLIBS")
-file( WRITE  ${SCRIPT_NAME_TCSH} "#/bin/tcsh\n")
-file( APPEND ${SCRIPT_NAME_TCSH} "\n")
-file( APPEND ${SCRIPT_NAME_TCSH} "echo 'Setting environment variables for NCEPLIBS and its dependencies'\n")
-file( APPEND ${SCRIPT_NAME_TCSH} "# Dependencies\n")
-file( APPEND ${SCRIPT_NAME_TCSH} "setenv PATH \"${EXTERNAL_LIBS_DIR}/${CMAKE_INSTALL_BINDIR}:\$\{PATH\}\"\n")
-file( APPEND ${SCRIPT_NAME_TCSH} "if ( $?LD_LIBRARY_PATH ) then\n")
-file( APPEND ${SCRIPT_NAME_TCSH} "  setenv LD_LIBRARY_PATH \"${EXTERNAL_LIBS_DIR}/${CMAKE_INSTALL_LIBDIR}:\$\{LD_LIBRARY_PATH\}\"\n")
-file( APPEND ${SCRIPT_NAME_TCSH} "else\n")
-file( APPEND ${SCRIPT_NAME_TCSH} "  setenv LD_LIBRARY_PATH \"${EXTERNAL_LIBS_DIR}/${CMAKE_INSTALL_LIBDIR}\"\n")
-file( APPEND ${SCRIPT_NAME_TCSH} "endif\n")
-file( APPEND ${SCRIPT_NAME_TCSH} "setenv ESMFMKFILE ${ESMFMKFILE}\n")
-file( APPEND ${SCRIPT_NAME_TCSH} "setenv NETCDF ${NETCDF}\n")
-file( APPEND ${SCRIPT_NAME_TCSH} "# NCEPLIBS\n")
-file( APPEND ${SCRIPT_NAME_TCSH} "setenv NCEPLIBS_DIR ${CMAKE_INSTALL_PREFIX}\n")
-file( APPEND ${SCRIPT_NAME_TCSH} "setenv NEMSIO_INC ${CMAKE_INSTALL_PREFIX}/include\n")
-file( APPEND ${SCRIPT_NAME_TCSH} "setenv NEMSIO_LIB ${CMAKE_INSTALL_PREFIX}/lib/libnemsio_v2.2.3.a\n")
-file( APPEND ${SCRIPT_NAME_TCSH} "setenv BACIO_LIB4 ${CMAKE_INSTALL_PREFIX}/lib/libbacio_v2.1.0_4.a\n")
-file( APPEND ${SCRIPT_NAME_TCSH} "setenv SP_LIBd ${CMAKE_INSTALL_PREFIX}/lib/libsp_v2.0.2_d.a\n")
-file( APPEND ${SCRIPT_NAME_TCSH} "setenv W3EMC_LIBd ${CMAKE_INSTALL_PREFIX}/lib/libw3emc_v2.2.0_d.a\n")
-file( APPEND ${SCRIPT_NAME_TCSH} "setenv W3NCO_LIBd ${CMAKE_INSTALL_PREFIX}/lib/libw3nco_v2.0.6_d.a\n")
-file( APPEND ${SCRIPT_NAME_TCSH} "#\n")
+message(STATUS "Creating shell scripts to set environment variables for using NCEPLIBS")
+configure_file(setenv_nceplibs_csh_template.txt ${CMAKE_INSTALL_PREFIX}/bin/setenv_nceplibs.csh @ONLY)
```

Comment:
The only question I have here is if this is safe without path. I assume so, because CMakeLists.txt lives in the same directory as the setenv templates, and it works for me on all the systems I used.